### PR TITLE
revised logic of histogram constructor

### PR DESCRIPTION
**Pull request by rnugent3**

### CODE REVIEW

File: Statistics/Histograms/Histogram.cs [hunk 1]
```diff
@@ -115,10 +115,7 @@ public Histogram(double[] data, double binWidth)
             _BinWidth = binWidth;
             if (data == null)
             {
-                Min = 0;
-                Max = Min + _BinWidth;
-                Int64 numberOfBins = 1;
-                _BinCounts = new Int32[numberOfBins];
+                throw new ArgumentNullException("The histogram cannot be constructed without data. Use a different constructor.");
```
Comment (HenryGeorgist) [inline Statistics/Histograms/Histogram.cs:118]: i dont think we need to throw the argumentnullexception. i would rather not create exceptions in construtors. id change the if to if data!=null, do what is in the else case right now, else, dont do anything.

Comment (HenryGeorgist) [reply Statistics/Histograms/Histogram.cs:118]: we might set _bincounts = new int32[]

File: StatisticsTests/Histograms/HistogramTests.cs [hunk 2]
```diff
@@ -211,7 +211,8 @@ public void NormallyDistributed_Histogram_CentralTendency(int n, double binWidth
         [InlineData(.1,1)]
         public void IsHistogramConstructableWithNullData(double binWidth, double expected)
         {
-            Histogram histogram = new Histogram(null, binWidth);
+            double observationValue = 0.5;
+            Histogram histogram = new Histogram(observationValue, binWidth);
```
Comment (HenryGeorgist) [inline StatisticsTests/Histograms/HistogramTests.cs:215]: this is no longer using histogram with null data, i think we set this back to null right?

**Review by HenryGeorgist — CHANGES_REQUESTED** [review]
i think what we do in the test varies now from what the name of the test is - that seems wrong, i think we can have the argument be null, but i dont think we should throw the exception.

File: StatisticsTests/Histograms/HistogramTests.cs (hunk 2 above)
Comment (rnugent3) [reply StatisticsTests/Histograms/HistogramTests.cs:215]: the histogram should not be constructed using null data with the Histogram(data, binwidth) constructor, right? that doesn't really make sense to me. I'd rather delete the test.

File: Statistics/Histograms/Histogram.cs (hunk 1 above)
Comment (rnugent3) [reply Statistics/Histograms/Histogram.cs:118]: done

**Comment by rnugent3** [comment]
image

**Comment by rnugent3** [comment]
@HenryGeorgist any thoughts? I have not touched any logic outside the constructor, and made the changes as you suggested. I am not sure why my changes would make your test fail?

**Comment by rnugent3** [comment]
weird...the test fails on my machine but passes on GitHub, the converse of what I was experiencing earlier this week.

File: StatisticsTests/Histograms/HistogramTests.cs (hunk 2 above)
Comment (HenryGeorgist) [reply StatisticsTests/Histograms/HistogramTests.cs]: yeah i see what you are saying, i dont disagree. but if we detect data is null, it is constructed with the binwidth set. as long as we dont call "min or max" on a null array we havent corrupted data nor created an improper data behavior, nor triggered a null pointer. id rather drop the array constructor all together.

**Comment by HenryGeorgist** [comment]
yeah and you didnt even change the constructor being used in the test. i dont get it

**Review by HenryGeorgist — APPROVED** [review]
i think this is good to go.

**Pull request merged**